// react-accessibility-widget/dist/context/AccessibilityContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useAccContext = exports.AccessibilityProvider = exports.AccessibilityContext = void 0;
var _react = _interopRequireWildcard(require("react"));
var _domUtils = require("../utils/domUtils");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
// ✅ Import utility function

const AccessibilityContext = exports.AccessibilityContext = /*#__PURE__*/(0, _react.createContext)();
const AccessibilityProvider = _ref => {
  let {
    children
  } = _ref;
  const [settings, setSettings] = (0, _react.useState)(() => {
    const savedSettings = JSON.parse(localStorage.getItem("accessibilitySettings"));
    return savedSettings || {
      fontSize: 16,
      lineHeight: 1.5,
      contrast: 100,
      brightness: 100,
      saturation: 100,
      greyscale: 0,
      invertColors: false,
      dyslexicFont: false,
      hideImages: false,
      highlightTitles: false,
      highlightLinks: false,
      cursorSize: "default",
      animationEnabled: true,
      readingLine: false,
      readingMask: false
    };
  });
  (0, _react.useEffect)(() => {
    (0, _domUtils.applyAccessibilityStyles)(settings);
  }, []);
  (0, _react.useEffect)(() => {
    localStorage.setItem("accessibilitySettings", JSON.stringify(settings));
    (0, _domUtils.applyAccessibilityStyles)(settings);
  }, [settings]);
  (0, _react.useEffect)(() => {
    if (!settings.readingLine) {
      document.querySelector("#readingLine")?.remove();
    }
  }, [settings.readingLine]);
  const updateSetting = (key, value) => {
    setSettings(prev => ({
      ...prev,
      [key]: value
    }));
  };
  return /*#__PURE__*/_react.default.createElement(AccessibilityContext.Provider, {
    value: {
      settings,
      updateSetting,
      setSettings
    }
  }, children);
};
exports.AccessibilityProvider = AccessibilityProvider;
const useAccContext = () => (0, _react.useContext)(AccessibilityContext);
exports.useAccContext = useAccContext;
