// react-accessibility-widget/dist/AccessibilityWidget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _FontSizeControl = _interopRequireDefault(require("./components/FontSizeControl"));
var _ContrastControl = _interopRequireDefault(require("./components/ContrastControl"));
var _InvertColors = _interopRequireDefault(require("./components/InvertColors"));
var _HideImagesControl = _interopRequireDefault(require("./components/HideImagesControl"));
var _DyslexicFontToggle = _interopRequireDefault(require("./components/DyslexicFontToggle"));
var _BrightnessControl = _interopRequireDefault(require("./components/BrightnessControl"));
var _AnimationControl = _interopRequireDefault(require("./components/AnimationControl"));
var _CursorControl = _interopRequireDefault(require("./components/CursorControl"));
require("./styles/accessibility.css");
var _GreyscaleMode = _interopRequireDefault(require("./components/GreyscaleMode"));
var _dummyImage = _interopRequireDefault(require("./assets/icons/dummy-image.png"));
var _HighlightLinks = _interopRequireDefault(require("./components/HighlightLinks"));
var _HighlightTitles = _interopRequireDefault(require("./components/HighlightTitles"));
var _LineHeightControl = _interopRequireDefault(require("./components/LineHeightControl"));
var _ReadingMask = _interopRequireDefault(require("./components/ReadingMask"));
var _SaturationControl = _interopRequireDefault(require("./components/SaturationControl"));
var _ReadingLine = _interopRequireDefault(require("./components/ReadingLine"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const AccessibilityWidget = () => {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "accessibility-widget"
  }, /*#__PURE__*/_react.default.createElement("h3", {
    style: {
      color: "#056BCA"
    }
  }, "Accessibility Settings"), /*#__PURE__*/_react.default.createElement("h2", null, "Accessibility Settings"), " ", /*#__PURE__*/_react.default.createElement("h1", null, "Accessibility Settings"), " ", /*#__PURE__*/_react.default.createElement("h4", null, "Accessibility Settings"), /*#__PURE__*/_react.default.createElement(_FontSizeControl.default, null), /*#__PURE__*/_react.default.createElement(_ContrastControl.default, null), /*#__PURE__*/_react.default.createElement(_AnimationControl.default, null), /*#__PURE__*/_react.default.createElement(_CursorControl.default, null), /*#__PURE__*/_react.default.createElement(_GreyscaleMode.default, null), /*#__PURE__*/_react.default.createElement(_HighlightLinks.default, null), /*#__PURE__*/_react.default.createElement(_InvertColors.default, null), /*#__PURE__*/_react.default.createElement(_LineHeightControl.default, null), /*#__PURE__*/_react.default.createElement(_ReadingLine.default, null), /*#__PURE__*/_react.default.createElement(_ReadingMask.default, null), /*#__PURE__*/_react.default.createElement(_HighlightTitles.default, null), /*#__PURE__*/_react.default.createElement(_HideImagesControl.default, null), /*#__PURE__*/_react.default.createElement(_DyslexicFontToggle.default, null), /*#__PURE__*/_react.default.createElement(_BrightnessControl.default, null), /*#__PURE__*/_react.default.createElement(_SaturationControl.default, null), /*#__PURE__*/_react.default.createElement("img", {
    src: _dummyImage.default,
    alt: "dmmy imges"
  }), /*#__PURE__*/_react.default.createElement("a", {
    href: "#"
  }, "link me"), /*#__PURE__*/_react.default.createElement("p", null, "jbfjbdsfjkfjbjkbkkbsbfdsfdsfdsfdsfdsgdgdfgdfgdfgdfgdfgfgdncvbdxfsgdsfdfgdf gdgdfgdfgdfgdfgdsfsdfdsfsdfsdfdsfdsfdsfdsffsdfdsfdsfsdfdsfdsfds dfgdfgdsfgdsfgdsfdsfdsfsdfsdfsdfdsfdsfdsdfsfdsfdsfdsffsdfdsfdsf dfdfdsfdsfsfdsfdsfdsfdsfsdfsdfdsfdsfsdfdsfsddsfsdfdsfsddsfdsfds"));
};
var _default = exports.default = AccessibilityWidget;
