// react-accessibility-widget/src/components/ReadingLine.js
import React, { useState, useEffect } from "react";
import { useAccContext } from "../context/AccessibilityContext";

const ReadingLine = () => {
  const { settings, updateSetting } = useAccContext();
  const [position, setPosition] = useState(0);

  useEffect(() => {
    if (settings.readingLine) {
      const handleMouseMove = (event) => {
        setPosition(event.clientY);
      };
      document.addEventListener("mousemove", handleMouseMove);
      return () => document.removeEventListener("mousemove", handleMouseMove);
    }
  }, [settings.readingLine]);

  return (
    <div>
      <label>
        <input
          type="checkbox"
          checked={settings.readingLine}
          onChange={() => updateSetting("readingLine", !settings.readingLine)}
        />
        Enable Reading Line
      </label>

      {settings.readingLine && (
        <div
          style={{
            position: "fixed",
            top: `${position}px`,
            left: 0,
            width: "100%",
            height: "2px",
            backgroundColor: "yellow",
            pointerEvents: "none",
            zIndex: 9999,
          }}
        ></div>
      )}
    </div>
  );
};

export default ReadingLine;
