// react-accessibility-widget/src/context/AccessibilityContext.js
import { createContext, useState, useEffect, useContext } from "react";
import { applyAccessibilityStyles } from "../utils/domUtils"; // ✅ Import utility function
import React from 'react';

export const AccessibilityContext = createContext();

export const AccessibilityProvider = ({ children }) => {
  const [settings, setSettings] = useState(() => {
    const savedSettings = JSON.parse(localStorage.getItem("accessibilitySettings"));
    return savedSettings || {
      fontSize: 16,
      lineHeight: 1.5,
      contrast: 100,
      brightness: 100,
      saturation: 100,
      greyscale: 0,
      invertColors: false,
      dyslexicFont: false,
      hideImages: false,
      highlightTitles: false,
      highlightLinks: false,
      cursorSize: "default",
      animationEnabled: true,
      readingLine: false,
      readingMask: false,
    };
  });

  useEffect(() => {
    applyAccessibilityStyles(settings);
  }, []);

  useEffect(() => {
    localStorage.setItem("accessibilitySettings", JSON.stringify(settings));
    applyAccessibilityStyles(settings);
  }, [settings]);

  useEffect(() => {
    if (!settings.readingLine) {
      document.querySelector("#readingLine")?.remove();
    }
  }, [settings.readingLine]);
  

  const updateSetting = (key, value) => {
    setSettings((prev) => ({ ...prev, [key]: value }));
  };

  return (
    <AccessibilityContext.Provider value={{ settings, updateSetting, setSettings }}>
      {children}
    </AccessibilityContext.Provider>
  );
};

export const useAccContext = () => useContext(AccessibilityContext);
