// react-accessibility-widget/dist/components/CursorControl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _useAccessibility = _interopRequireDefault(require("../hooks/useAccessibility"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const CursorControl = () => {
  const {
    settings,
    updateSetting
  } = (0, _useAccessibility.default)();
  const handleCursorChange = event => {
    updateSetting("cursorSize", event.target.value);
  };
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("label", null, "Cursor Size"), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("input", {
    type: "radio",
    id: "defaultCursor",
    name: "cursor",
    value: "default",
    checked: settings.cursorSize === "default",
    onChange: handleCursorChange
  }), /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "defaultCursor"
  }, "Default")), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("input", {
    type: "radio",
    id: "bigBlackCursor",
    name: "cursor",
    value: "big-black",
    checked: settings.cursorSize === "big-black",
    onChange: handleCursorChange
  }), /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "bigBlackCursor"
  }, "Big Black Cursor")), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("input", {
    type: "radio",
    id: "bigWhiteCursor",
    name: "cursor",
    value: "big-white",
    checked: settings.cursorSize === "big-white",
    onChange: handleCursorChange
  }), /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "bigWhiteCursor"
  }, "Big White Cursor")));
};
var _default = exports.default = CursorControl;
