// react-accessibility-widget/dist/components/LineHeightControl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _AccessibilityContext = require("../context/AccessibilityContext");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const LineHeightControl = () => {
  const {
    settings,
    updateSetting
  } = (0, _AccessibilityContext.useAccContext)();
  const handleChange = event => {
    const newLineHeight = event.target.value;
    updateSetting("lineHeight", newLineHeight);
  };
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "lineHeightControl"
  }, "Line Height"), /*#__PURE__*/_react.default.createElement("input", {
    type: "range",
    id: "lineHeightControl",
    min: "1",
    max: "3",
    step: "0.1",
    value: settings.lineHeight,
    onChange: handleChange
  }), /*#__PURE__*/_react.default.createElement("span", null, settings.lineHeight));
};
var _default = exports.default = LineHeightControl;
