// react-accessibility-widget/dist/components/ReadingLine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _AccessibilityContext = require("../context/AccessibilityContext");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const ReadingLine = () => {
  const {
    settings,
    updateSetting
  } = (0, _AccessibilityContext.useAccContext)();
  const [position, setPosition] = (0, _react.useState)(0);
  (0, _react.useEffect)(() => {
    if (settings.readingLine) {
      const handleMouseMove = event => {
        setPosition(event.clientY);
      };
      document.addEventListener("mousemove", handleMouseMove);
      return () => document.removeEventListener("mousemove", handleMouseMove);
    }
  }, [settings.readingLine]);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("label", null, /*#__PURE__*/_react.default.createElement("input", {
    type: "checkbox",
    checked: settings.readingLine,
    onChange: () => updateSetting("readingLine", !settings.readingLine)
  }), "Enable Reading Line"), settings.readingLine && /*#__PURE__*/_react.default.createElement("div", {
    style: {
      position: "fixed",
      top: `${position}px`,
      left: 0,
      width: "100%",
      height: "2px",
      backgroundColor: "yellow",
      pointerEvents: "none",
      zIndex: 9999
    }
  }));
};
var _default = exports.default = ReadingLine;
