// react-accessibility-widget/dist/components/ReadingMask.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _AccessibilityContext = require("../context/AccessibilityContext");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const ReadingMask = () => {
  const {
    settings,
    updateSetting
  } = (0, _AccessibilityContext.useAccContext)();
  const [maskPosition, setMaskPosition] = (0, _react.useState)(0);
  const handleToggle = () => {
    updateSetting("readingMask", !settings.readingMask);
  };
  (0, _react.useEffect)(() => {
    if (settings.readingMask) {
      const handleMouseMove = event => {
        setMaskPosition(event.clientY - 50); // Center mask around cursor
      };
      window.addEventListener("mousemove", handleMouseMove);
      return () => window.removeEventListener("mousemove", handleMouseMove);
    }
  }, [settings.readingMask]);
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "readingMaskToggle"
  }, "Enable Reading Mask", /*#__PURE__*/_react.default.createElement("input", {
    type: "checkbox",
    id: "readingMaskToggle",
    checked: settings.readingMask || false,
    onChange: handleToggle
  })), settings.readingMask && /*#__PURE__*/_react.default.createElement("div", {
    className: "reading-mask",
    style: {
      top: `${maskPosition}px`
    }
  }));
};
var _default = exports.default = ReadingMask;
