// react-accessibility-widget/dist/components/SaturationControl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _AccessibilityContext = require("../context/AccessibilityContext");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const SaturationControl = () => {
  const {
    settings,
    updateSetting
  } = (0, _AccessibilityContext.useAccContext)();
  const handleSaturationChange = event => {
    const newSaturation = event.target.value;
    updateSetting("saturation", newSaturation);
    document.body.style.filter = `saturate(${newSaturation}%)`;
  };
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("label", {
    htmlFor: "saturationRange"
  }, "Saturation"), /*#__PURE__*/_react.default.createElement("input", {
    id: "saturationRange",
    type: "range",
    min: "0",
    max: "200",
    value: settings.saturation,
    onChange: handleSaturationChange
  }));
};
var _default = exports.default = SaturationControl;
